include/boost/algorithm/apply_permutation.hpp

```diff
@@ -39,17 +47,21 @@ namespace boost { namespace algorithm
 template<typename RandomAccessIterator1, typename RandomAccessIterator2>
 void
 apply_permutation(RandomAccessIterator1 item_begin, RandomAccessIterator1 item_end,
-                  RandomAccessIterator2 ind_begin, RandomAccessIterator2 ind_end)
+                  RandomAccessIterator2 ind_begin, RandomAccessIterator2)
```

Comment:
I have removed the fourth iterator identifier because, being unused, it causes warnings.